CoreInteractiveGraph/src/au/gov/asd/tac/constellation/graph/interaction/visual/renderables/SelectionFreeformRenderable.java

```diff
@@ -0,0 +1,181 @@
+
+import au.gov.asd.tac.constellation.utilities.graphics.Matrix44f;
+import au.gov.asd.tac.constellation.utilities.graphics.Vector3f;
+import au.gov.asd.tac.constellation.utilities.graphics.Vector4f;
+import au.gov.asd.tac.constellation.visual.opengl.renderer.GLRenderable;
+import au.gov.asd.tac.constellation.visual.opengl.renderer.GLVisualProcessor;
+import au.gov.asd.tac.constellation.visual.opengl.renderer.batcher.Batch;
+import au.gov.asd.tac.constellation.visual.opengl.utilities.GLTools;
+import au.gov.asd.tac.constellation.visual.opengl.utilities.ShaderManager;
+import com.jogamp.opengl.GL;
+import com.jogamp.opengl.GL3;
+import com.jogamp.opengl.GLAutoDrawable;
+import java.awt.Point;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.FloatBuffer;
+import java.util.concurrent.BlockingDeque;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.apache.commons.collections4.CollectionUtils;
+
+/**
+ * Draw a selection freeform shape.
+ *
+ * @author CrucisGamma
+ */
+public class SelectionFreeformRenderable implements GLRenderable {
+
+    // How many vertices do we need to draw a freeform shape?
+    // What color is the selection freeform shape?
+    private static final Vector4f SELECTION_COLOR = new Vector4f(0, 0.5f, 1, 0.375f);
+    private static final Vector4f SECOND_SELECTION_COLOR = new Vector4f(1, 0.5f, 0, 0.375f);
+    private static final Vector3f ZERO_3F = new Vector3f(0, 0, 0);
+    private int shader;
+    private int shaderMvp;
+    private final Batch batch;
+    private SelectionFreeformModel selectionFreeformModel;
+    private final BlockingDeque<SelectionFreeformModel> modelQueue = new LinkedBlockingDeque<>();
+
+    private static final int NUMBER_OF_VERTICES = 50;
+    private static final int COLOR_BUFFER_WIDTH = 4;
+    private static final int VERTEX_BUFFER_WIDTH = 3;
+
+    private final int colorTarget;
+    private final int vertexTarget;
+
+    public SelectionFreeformRenderable() {
+        batch = new Batch(GL3.GL_TRIANGLE_FAN);
+        vertexTarget = batch.newFloatBuffer(VERTEX_BUFFER_WIDTH, true);
+        colorTarget = batch.newFloatBuffer(COLOR_BUFFER_WIDTH, true);
+    }
+
+    @Override
+    public int getPriority() {
+        return RenderablePriority.ANNOTATIONS_PRIORITY.getValue();
+    }
+
+    @Override
+    public void init(final GLAutoDrawable drawable) {
+        final GL3 gl = drawable.getGL().getGL3();
+        String vs = null;
+        String gs = null;
+        String fs = null;
+
+        try {
+            vs = GLTools.loadFile(GLVisualProcessor.class, "shaders/PassThru.vs");
+            gs = GLTools.loadFile(GLVisualProcessor.class, "shaders/PassThruTriangle.gs");
+            fs = GLTools.loadFile(GLVisualProcessor.class, "shaders/PassThru.fs");
+        } catch (IOException ex) {
+            Logger.getLogger(SelectionFreeformRenderable.class.getName()).log(Level.SEVERE, null, ex);
```

Comment:
I would extract the logger into a private static final field (see other classes for examples). This allows the logger to easily be used elsewhere in the class